xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -1650,34 +1420,27 @@ absl::StatusOr<bool> FuseBwdMultiHeadedAttentionBlock(
       ChangeCheckedDimToFastest(comp, bmm_2_grad_1, false /*is_lhs*/,
                                 false /*should_contracting_be_fastest*/));
   (void)bmm_2_grad_1_rhs;
-  // Operand order: {Q, K, V, Fwd act, d_o, mask*, bias*, O*}
+  // Operand order: {Q, K, V, Fwd act, d_o, bias*, O*}
   std::vector<HloInstruction*> operands = {
       rhs_bmm1_grad_gemm1, lhs_bmm1_grad_gemm2, rhs_bmm2_grad_gemm2, fwd_act,
       d_output_grad};
-  if (mask) {
-    HloInstruction* converted_mask = comp->AddInstruction(
-        HloInstruction::CreateConvert(bmm_2_grad_2->shape(), mask));
-    operands.push_back(converted_mask);
-  }
 
-  // if is flash attention, add fwd output to input list
-  if (is_flash_attention) {
-    if (!is_causal_mask && bias) {
-      operands.push_back(bias);
-    }
-    HloInstruction* fwd_output;
-    for (auto user : fwd_fmha_call->users()) {
-      if (user->opcode() == HloOpcode::kGetTupleElement &&
-          user->tuple_index() == 0) {
-        fwd_output = user;
-      }
+  // For flash attention, add fwd output to input list
+  if (!is_causal_mask && bias) {
```

Comment:
If mask is always combined with bias, should we always add bias here?

Reply:
is_causal_mask is used to indicated mask generation inside cuDNN. If the bias is causal mask, we don't take in bias, we just generate causal mask directly.

Reply:
ah right, thanks.